projects/js-packages/charts/src/components/pie-semi-circle-chart/pie-semi-circle-chart.tsx

```diff
@@ -109,6 +109,52 @@ const PieSemiCircleChart: FC< PieSemiCircleChartProps > = ( {
 		[ handleMouseMove ]
 	);
 
+	// Add validation check
+	const { isValid, message } = validateData( data );
+
+	if ( ! isValid ) {
+		return (
+			<div className={ styles[ 'pie-semi-circle-chart' ] }>
+				<svg width={ width } height={ width / 2 } data-testid="pie-chart-svg">
+					<text x="50%" y="50%" textAnchor="middle" className={ styles.error }>
+						{ message }
+					</text>
+				</svg>
+			</div>
+		);
+	}
+
+	const height = width / 2;
```

Comment:
these bits as well - if we add the `// Add validation check` about `handleMouseMove`, then the code comparing would work nicely. Now it's hard to parse.

Plus: I don't think removing `centerX` is an improvement regarding readability - we should limit the scope if we can.